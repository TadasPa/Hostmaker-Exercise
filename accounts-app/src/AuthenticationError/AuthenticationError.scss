@import "../constants.scss";

.error-message {
    font-size: 1rem;
    min-height: 5.6rem;
    max-width: 70rem;
    border-width: 0rem;
    background-color: $error-message-color;
    border-radius: 1rem;
    letter-spacing: 0.1rem;    
    margin-top: 2rem;
    margin-bottom: 2rem;
    color: #ffffff;
    display: flex;
    align-items: center;
    padding-left: 2rem;
    padding-right: 2rem;
}