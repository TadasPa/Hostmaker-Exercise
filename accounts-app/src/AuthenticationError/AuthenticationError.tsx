import React, { Component } from "react";
import "./AuthenticationError.scss";

export interface IAuthenticationErrorProps {
    error: string;
}

export class AuthenticationError extends Component<IAuthenticationErrorProps> {

    render() {
        return (
            this.props.error ? (<div className="error-message"> {this.props.error} </div>) : "" 
            );
    }
}

export default AuthenticationError; 