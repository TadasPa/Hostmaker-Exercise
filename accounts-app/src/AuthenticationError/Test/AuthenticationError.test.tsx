import React from 'react';
import { shallow } from 'enzyme';
import { AuthenticationError } from '../AuthenticationError';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    shallow(<AuthenticationError error={""} />);
});


test('Error renders correctly when no error to show', () => {
    const spinner = renderer
        .create(<AuthenticationError error={""} />)
        .toJSON();
    expect(spinner).toMatchSnapshot();
});

test('Error renders correctly when there is error to show', () => {
    const spinner = renderer
        .create(<AuthenticationError error={"Error, please try again."} />)
        .toJSON();
    expect(spinner).toMatchSnapshot();
});