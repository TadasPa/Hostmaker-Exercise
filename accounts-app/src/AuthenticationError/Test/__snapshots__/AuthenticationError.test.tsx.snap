// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly when no error to show 1`] = `""`;

exports[`Error renders correctly when there is error to show 1`] = `
<div
  className="error-message"
>
   
  Error, please try again.
   
</div>
`;
